File: backend/models/Interview.js
const mongoose = require('mongoose');

const interviewSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  topic: String,
  questions: [{
    question: String,
    userAnswer: String,
    score: Number,
    feedback: String
  }],
  totalScore: Number,
  totalQuestions: Number,
  completedAt: Date
}, { timestamps: true });

module.exports = mongoose.model('Interview', interviewSchema);

File: backend/models/interview.py
"""models/interview.py — Interview table (replaces Mongoose Interview.js)"""
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey, JSON
from sqlalchemy.sql import func
from database import Base


class Interview(Base):
    __tablename__ = "interviews"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    topic = Column(String(255), nullable=True)
    # questions: list of {question, userAnswer, score, feedback}
    questions = Column(JSON, default=list)
    total_score = Column(Float, nullable=True)
    total_questions = Column(Integer, nullable=True)
    completed_at = Column(DateTime(timezone=True), nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
